### ISSUE TITLE
Oxxo order remains on-hold even after payment is complete

### ISSUE THREAD

**Issue by james-allan** [woocommerce/woocommerce-gateway-stripe#3125]
**Describe the bug**

Discovered by GlobalStep in recent testing of the 8.3 release. 

After waiting the 3-5 minutes for Stripe to process Oxxo test transactions, the payment is successful and Stripe sends the webhook, however the order is left on-hold. 

Looking in the Stripe plugin logs the following error:

```
2024-05-07T04:46:14+00:00 Debug 
====Stripe Version: 8.3.0====
====Stripe Plugin API Version: 2019-09-09====
====Start Log====
Could not find order via charge ID: py_3PDfruH8EetsbgDL1YtkbDp3
====End Log====
```

**To Reproduce**
1. Create a stripe account located in Mexico.
2. Set the API credentials and webhook in the plugin settings.
3. Set the store currency to MXN.
4. Enable Oxxo. 
5. Complete a purchase using Oxxo.
6. Wait the 5 minutes Stripe takes to process the test payment. 
7. Review the order in WC and notice the payment is on-hold. 
8. Confirm the order in Stripe is Complete. 

| WooCommerce | Stripe |
|--------|--------|
| <img width="809" alt="Screenshot 2024-05-07 at 3 00 38 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/24e449b7-dd0e-4a01-b12d-752931ba329f"> | <img width="456" alt="Screenshot 2024-05-07 at 3 00 49 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/3ee4b4ea-5c2b-4eee-ac4c-5f30d8ffc4ab"> | 

**Expected behavior**
The order should be processing/complete.

**Comment by james-allan** [comment]
This issue isn't present on 8.2 (current live version) so this has been introduced in recent changes.

**Comment by james-allan** [comment]
> Looking in the Stripe plugin logs the following error:

This is a red herring. I think this log entry also appears for successful payments. 


After a little more digging the issue appears to have been introduced in: https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3078

The following log entry is entered for the unsuccessful webhook processing while 8.3 is active. 

```
Stripe UPE waiting for redirect. The status for order 20 might need manual adjustment.
```

**Comment by james-allan** [comment]
After discussing with @mattallan in Slack, we agree that the right course of action here is to exclude voucher payment methods from the checks in the webhook processing. 

> So that meta check you added (in #3078) was designed to prevent webhooks processing the payment right as the redirect payment processing happens. Given webhook processing is the only method of payment processing for these [vouchers], then it should be excluded by any checks like that.

**Comment by glagonikas** [comment]
@james-allan we're experiencing this on 8.3.0 using card payments. This also results in the subscriptions remaining on-hold. Will the proposed PR resolve the issue and when will it be released?

update: i can see the PR is already merged on the code that shipped with 8.3.0 but we're still experiencing this issue

**Comment by james-allan** [comment]
No, this issue shouldn't have affected card payments. The issue was specific to Oxxo and Boleto which are voucher payment methods. The issue was caused by an issue in webhook processing and Oxxo and Boleto are exclusively processed via webhooks. 

IIRC cards should be processed immediately via API requests so I'm not sure why you'd have card payments stuck on-hold unless you're using authorize and capture.
